### ISSUE TITLE
Not using VGG16?

### ISSUE THREAD

**Issue by MarineBiologist272** [AdrianNunez/Fall-Detection-with-CNNs-and-Optical-Flow#49]
Looking at the code tf2_temporalnet_urfd3.py, the model for VGG16 is no longer used after line 376: model.compile(optimizer=adam, loss='categorical_crossentropy',metrics=['accuracy'])

In fact, the code still runs when commenting out the entire VGG16 section, as well as any lines involving model, and the accuracy is (so far from my testing) unaffected!

**Comment by AdrianNunez** [comment]
Hello,

The VGG16 (or, better said, its convolutional layers) is only used for feature extraction. The model you are referring to is the actual classifier (an MLP). I guess the results do not change because you are using saved features, so the VGG16 has already done its job. I believe line 366 is of no use and it is not necessary, sorry for that. Line 356 should be the last one in which the VGG16 model is used. I hope this solves the issue.

**Comment by MarineBiologist272** [comment]
Ah I see.
I noticed that the VG16 network takes in a (224, 224, 20) matrix. Could you explain how that is done from grayscale optical flow jpg which you provided?

**Comment by AdrianNunez** [comment]
This is explained in [the paper](https://www.hindawi.com/journals/wcmc/2017/9474806/). Basically, grayscale OF images are stacked.

**Comment by MarineBiologist272** [comment]
Thank you so much. I have read that paper and understand that they are stacked. It also states that the frames are consecutive. However, the number of pictures in a uefd video folder >> 20. Am I correct to say you used a sliding window and breaked the video into multiple parts of 20 frames?

**Comment by AdrianNunez** [comment]
Yes, I used an sliding window approach and have several stacks per video. I believe these details are also contained in the paper.
